src/app/public/theming/theme-if.directive.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Directive,
+  Host,
+  Input,
+  OnDestroy,
+  Optional,
+  TemplateRef,
+  ViewContainerRef
+} from '@angular/core';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  SkyThemeSettings
+} from './theme-settings';
+
+import {
+  SkyThemeService
+} from './theme.service';
+
+import {
+  SkyThemeDirective
+} from './theme.directive';
+
+/**
+ * Component that works like `ngIf` to show markup for matching theme.
+ *
+ * If the directive is within a `skyTheme` directive, it uses settings from that directive.
+ */
+@Directive({
+  selector: '[skyThemeIf]'
+})
+export class SkyThemeIfDirective implements OnDestroy {
+  private context: string;
+  private currentTheme: SkyThemeSettings | undefined;
+  private ngUnsubscribe = new Subject();
+  private hasView: boolean = false;
+
+  constructor(
+    @Optional() @Host() public skyTheme: SkyThemeDirective,
+    private themeSvc: SkyThemeService,
+    private templateRef: TemplateRef<any>,
+    private viewContainer: ViewContainerRef
+  ) {
+    if (skyTheme) {
+      this.themeSettings = skyTheme.skyTheme;
+    }
+    this.themeSvc.settingsChange
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(settingsChange => {
+        this.themeSettings = settingsChange.currentSettings;
+      });
+  }
+
+  public ngOnDestroy(): void {
+    this.ngUnsubscribe.next();
+    this.ngUnsubscribe.complete();
+  }
+
+  /**
+   * A string that should match the name of a theme, such as `'default'` or `'modern'`.
+   *
+   * @param themeName
+   */
+  @Input()
+  set skyThemeIf(themeName: string) {
+    this.context = themeName.toLowerCase();
```

Comment:
Since you're calling `toLowerCase()` on `themeName`, you'll need to check it for `undefined` which you can do with the Elvis operator. Personally I don't really think we need to lower-case the value; we'd want consumers to specify the proper casing since elsewhere doing direct comparisons without our intervention will have to be done in a case-sensitive way. I'd also recommend changing the property type from `string` to `'default' | 'modern'` for extra type safety as well as for the benefit of the documentation generator.

```
this.context = themeName?.toLowerCase();
```